test/e2e/util_gatewayapi_test.go

```diff
@@ -189,13 +189,13 @@ func createHttpRoute(namespace, routeName, parentNamespace, hostname, backendRef
 
 	// Create the backend (service and pod) needed for the route to have resolvedRefs=true.
 	// The http route, service, and pod are cleaned up when the namespace is automatically deleted.
-	// buildEchoPod builds a pod that listens on port 8080.
-	echoPod := buildEchoPod(backendRefname, namespace)
-	if err := kclient.Create(context.TODO(), echoPod); err != nil {
-		return nil, fmt.Errorf("failed to create pod %s/%s: %v", namespace, echoPod.Name, err)
+	// buildEchoReplicaSet builds a replicaset which creates a pod that listens on port 8080.
+	echoRs := buildEchoReplicaSet(backendRefname, namespace)
+	if err := kclient.Create(context.TODO(), echoRs); err != nil {
```

Comment:
same comment as above on the create and retry. I am wondering if we care on actually creating some helper function for these "create" operations, with something like "create or retry" to avoid network issues.